anyway/widgets/suburban_widgets/AccidentCountByDriverTypeWidget.py

```diff
@@ -0,0 +1,84 @@
+import logging
+from collections import defaultdict
+from typing import Dict
+
+from flask_babel import _
+
+from anyway.RequestParams import RequestParams
+from anyway.backend_constants import DriverType
+from anyway.widgets.widget_utils import get_accidents_stats, get_injured_filters
+from anyway.models import InvolvedMarkerView
+from anyway.vehicle_type import VehicleCategory
+from anyway.widgets.Widget import register
+from anyway.widgets.suburban_widgets.SubUrbanWidget import SubUrbanWidget
+
+
+@register
+class AccidentCountByDriverTypeWidget(SubUrbanWidget):
+    name: str = "accident_count_by_driver_type"
+
+    def __init__(self, request_params: RequestParams):
+        super().__init__(request_params, type(self).name)
+        self.rank = 16
+        self.information = "Driver involvement in accident by driver type: professional - trucks, taxi, bus, work, minibus, tractor, private - private car, motorcycle, light electric - electric scooter, mobility scooter, electric bike."
+
+    def generate_items(self) -> None:
+        self.items = AccidentCountByDriverTypeWidget.count_accidents_by_driver_type(
+            self.request_params
+        )
+
+    @staticmethod
+    def count_accidents_by_driver_type(request_params):
+        involved_by_vehicle_type_data = get_accidents_stats(
+            table_obj=InvolvedMarkerView,
+            filters=get_injured_filters(request_params.location_info),
+            group_by="involve_vehicle_type",
+            count="involve_vehicle_type",
+            start_time=request_params.start_time,
+            end_time=request_params.end_time,
+        )
+        driver_types = defaultdict(int)
+        for item in involved_by_vehicle_type_data:
+            vehicle_type, count = item["involve_vehicle_type"], int(item["count"])
+            if vehicle_type in VehicleCategory.PROFESSIONAL_DRIVER.get_codes():
+                driver_types[
+                    # pylint: disable=no-member
+                    DriverType.PROFESSIONAL_DRIVER.get_label()
+                ] += count
+            elif vehicle_type in VehicleCategory.PRIVATE_DRIVER.get_codes():
+                driver_types[
+                    # pylint: disable=no-member
+                    DriverType.PRIVATE_VEHICLE_DRIVER.get_label()
+                ] += count
+            elif (
+                vehicle_type in VehicleCategory.LIGHT_ELECTRIC.get_codes()
+                or vehicle_type in VehicleCategory.OTHER.get_codes()
+            ):
+                # pylint: disable=no-member
+                driver_types[DriverType.OTHER_DRIVER.get_label()] += count

```

Comment:
here and in other places, simply getting the key, and only the updating, would give a cleaner code.

Reply:
I totally agree, the thing is - this and other places is existing code, I simply copy-pasted it. I think it is out of scope for this PR to refactor all the code, but I think that it is definitely an effort we should pursue at some point, hopefully sooner than later